app/controllers/projects_controller.rb

```diff
@@ -4,15 +4,18 @@
 # License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
 ###
 
-class ProjectsController < ApplicationController
-  before_action :require_can_view_projects!
-  before_action :require_can_delete_projects_or_data_sources!, only: [:destroy]
-  before_action :require_can_edit_projects!, only: [:edit, :update]
+class ProjectsController < ApplicationControllerV2
+  include ArelHelper
+
   before_action :set_project, only: [:show, :update, :edit, :destroy]
+  authorize_with { project_policy.can_view? }
```

Comment:
I think this is fine since we don't have an `index` page for projects currently, but `project_policy.can_view?` isn't the same as "am I allowed to view any projects?"  Do we maybe need a special case in `permission_granted_by_role?(permission)` when the resource id is not present?

Reply:
It's a good question with this pattern. My feeling is that can_index  doesn't fit well on a resource policy since it's a permission on a collection, not a single resource.

Often there is another entity that is a natural fit for a collection permission. For example on an enrollment index page, you'd probably have a project so you would check project_policy.can_view_enrollments

In this case, with projects#index, we just want to know if the user can just see the index page itself. I would say there's an implicit global 'site' or 'warehouse' singleton.resource you could ask. I might add a new global WarehousePolicy for those permissions. Then the checks would be something like this:
```ruby
  authorize_with { global_policy.can_view_projects? }, only: :index # equiv. to require_can_view_projects!
  authorize_with { project_policy.can_view? }, only: [:show, :update, :edit, :destroy]

  # or maybe if one global policy is too much, you could have a policy for each, for example a ProjectDashboardPolicy. Not sure...
  authorize_with { project_dashboard_policy.can_index? }

```

Reply:
Yeah, I keep trying to conflate the two.  Let's leave it as is for now, as a scope on whatever model is appropriate, but I think I like the idea of a warehouse authorization (or global authorization singleton) for these in the future.